python/dgl/nn/pytorch/explain/subgraphxexplainer.py

```diff
@@ -0,0 +1,566 @@
+        # Find and sort graph components by size from largest to smallest and take the biggest one.
+        biggest_comp = list(
+            sorted(nx.connected_components(nx_graph), key=len, reverse=True)
+        )[0]
+        # Convert back to DGLGraph object.
+        return list(biggest_comp)
+
+    def prune_graph(self, graph, high2low=True):
+        r"""Find the graph based on the chosen strategy. Once prunes, return the
+        list of subgraphs, list of subgraph nodes and list of pruned nodes.
+
+        Parameters
+        ----------
+        graph: DGLGraph
+            A homogeneous graph.
+        high2low: bool, optional
+            The strategy based on which the pruning will happen, "High2low" or "Low2high". If set
+            True, then "High2low" strategy will be followed.
+
+        Returns
+        -------
+        tuple(list, list, list)
+            Return a tuple consists of list of subgraphs based on pruned action,
+            list of subgraphs nodes, and list of pruned nodes.
+
+        """
+        subgraphs = []
+        subgraphs_nodes_mapping = []
+
+        out_degrees = graph.out_degrees()
+        in_degrees = graph.in_degrees()
+        nodes = graph.nodes()
+        node_degree = []
+
+        for node, in_degree, out_degree in zip(nodes, in_degrees, out_degrees):
+            node_degree.append((node, in_degree + out_degree))
+
+        node_degree = sorted(
+            node_degree, key=lambda x: x[1], reverse=bool(high2low)
+        )
+
+        if len(node_degree) < self.num_child_expand:
+            pruned_nodes = nodes
+        else:
+            pruned_nodes = [
+                node_degree[i][0] for i in range(self.num_child_expand)
+            ]
+
+        for prune_node in pruned_nodes:
+            new_subgraph_nodes = np.array(
+                [node for node in nodes if node != prune_node]
+            )
+            new_subgraph = dgl.node_subgraph(graph, new_subgraph_nodes)
+            biggest_comp = self.biggest_weak_component(new_subgraph)
+            new_subgraph_big_comp = dgl.node_subgraph(graph, biggest_comp)
+
+            subgraphs_nodes_mapping.append(new_subgraph_nodes[biggest_comp])
+            subgraphs.append(new_subgraph_big_comp)
+
+        return subgraphs, subgraphs_nodes_mapping, pruned_nodes
+
+    def explain_graph(self, graph, feat, **kwargs):
+        r"""Find the subgraph that play a crucial role to explain the prediction made
+        by the GNN for a graph.
+
+        Parameters
+        ----------
+        graph: DGLGraph
+            A homogeneous graph.
+        feat: Tensor
+            The input feature of shape :math:`(N, D)`. :math:`N` is the
+            number of nodes, and :math:`D` is the feature size.
+        kwargs: dict
+            Additional arguments passed to the GNN model. Tensors whose
+            first dimension is the number of nodes or edges will be
+            assumed to be node/edge features.
```

Comment:
"Additional arguments passed to the GNN model." is sufficient. The rest sentences do not hold in this case.

Reply:
done.